### ISSUE TITLE
disabling andOTP automatic android backup

### ISSUE THREAD

**Issue by Rendi-boi** [andOTP/andOTP#620]
Sorry if i shouldn't write this here, i don't know where to ask and i am not sure if this a bug or an unwanted feature.

**Is your feature request related to a problem? Please describe.**
I made a factory reset of my phone recently. And to my surprise after reinstalling apps, andOTP still has my vault how i left it. _Enable android sync_ is disabled in the andOTP settings, yet my vault was backuped to google drive, which i really don't want even if it is encryoted and all. 

**Describe the solution you'd like**
If possible, a clear indication whether user's data will be backed up automatically by android backup service or not.

**Comment by RichyHBM** [comment]
Had you ever enabled the google sync feature? It may have backed up at some point and never deleted since then?
We could maybe make it upload blank data when that toggle is off (if its not working as expected)

I do think that by default this still be enabled as it is a good security feature, in the past we had spoken of just having this be on always, as the user would expectedly want it to work if they have android sync enabled globally.

Just to confirm, your data is never uploaded to the cloud in a non-encrypted format!

**Comment by Rendi-boi** [comment]
Thank you. I don't think i enabled it ever manually unless the app did it by default after installation and i unchecked it later. 
And yeah maybe i overreacted about that when i saw my tokens after revovery like i left them and thought that such important data was somehow exposed while being in the cloud. Doesn't sound that bad when i think about it.

**Comment by RichyHBM** [comment]
So looking into https://github.com/aosp-mirror/platform_frameworks_base/blob/6bebb8418ceecf44d2af40033870f3aabacfe36e/core/java/android/app/backup/BackupAgentHelper.java our implementation of the BackupAgentHelper is correct, thus the only other options I can think of is that you had sync backup enabled at some point, or getAndroidBackupServiceEnabled() is always returning true no matter what.

We should add a log line, or toast in the settings to notify the user of the current value when the toggle is changed

**Comment by RichyHBM** [comment]
> Thank you. I don't think i enabled it ever manually unless the app did it by default after installation and i unchecked it later.
> And yeah maybe i overreacted about that when i saw my tokens after revovery like i left them and thought that such important data was somehow exposed while being in the cloud. Doesn't sound that bad now when i think about it.
> 
> Since it is pretty important, maybe user could be informed of the information about this feature at the first setup of the app with the option to enable or disable it at that moment? So that he could be aware that his data is going to be backed up automatically or that the backups are entirely up to him (with maybe a reminder that he should do them regulary just in case)

So yea, I'm thinking we add it as an option as part of the setup flow

**Comment by Rendi-boi** [comment]
I think, that sounds like a useful addition and should give a user some really important indications regarding the backups and security of such important data.
Thank you for giving me some understanding regarding this issue.

**Comment by KNejad** [comment]
I've just experienced this issue. I set up a new phone yesterday and have 0.9.0.1-play version on both devices. To my knowledge I've never enabled the Google Sync feature, yet somehow all my keys were already present on the new device. It had the latest version of the data available.

I double checked the toggle on the old phone and backup was not enabled.

Is there any way to check if the andOTP data is still on Google somehow and delete it manually? I'm afraid the data is still being synced without me realising
